src/main/java/be/syntra/devshop/DevshopBack/services/SearchServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import be.syntra.devshop.DevshopBack.models.SearchModel;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class SearchServiceImpl implements SearchService {
+    private final ProductService productService;
+
+    @Autowired
+    public SearchServiceImpl(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @Override
+    public List<Product> applySearchModel(SearchModel searchModel) {
+        List<Product> productList;
+
+        productList = (searchModel.isArchivedView())
+                ? productService.findAllByArchivedTrue() :
```

Comment:
mind the readability. Rather put the : on the next line, in front of the else-condition

Reply:
fixed